AutoResizeFont/DynamicLabel.swift

```diff
@@ -10,38 +10,60 @@ import UIKit
 
 @IBDesignable
 final class DynamicLabel: UILabel {
+    
+    //allow font to accept size from interface builder and scale up from there.
+    //this is to let us use values from the design specs
     @IBInspectable
-    var fontSizePts: CGFloat {
-        didSet {
-            calculateFontSize()
+    var pointSize = UIFont() {
+        didSet{
+            self.font = pointSize
+        }
+    }
+    
+    //override font for view
+    override var font : UIFont?{
+        didSet{
+            setNeedsLayout()
             setNeedsDisplay()
         }
     }
     
+    //deduced by multiplying the height of the Label by a constant
+    var relativeHeight : CGFloat!
+    
+    //relative constant to multiply our initial element height.
+    //this lets us double the size on bigger devices
+    let relativeFontConstant:CGFloat = 0.1
+    
     override init(frame: CGRect) {
-        fontSizePts = 14
-        super.init(frame: frame);
+        super.init(frame: frame)
+        relativeHeight = CGFloat(frame.height * relativeFontConstant)
```

Comment:
`frame.height` refers to the height of the label in question. Not sure that's a good reference point. A dev could assign any height to a label. Even if a dev doesn't do so, and the height is not fixed, then the height of the label would be dependent on the font size. This `frame.height` value should be the same for all devices as long as the font size is the same (which it is until the font is auto-resized, and the font's auto-resizing relies on this value).

Reply:
Do you think the algorithm used [here](https://stackoverflow.com/questions/53495725/how-to-ios-item-and-font-sizes-scaling-as-screen-size/53496831) will work? That code uses static font size increments for different device types, adding +2, +2, +3, +4 to the base font size depending on the device.

Reply:
the font relies on its label height, which is not much different from the font size measured in pts. I think it is a good point to reference because it will resize the font based on set autoLayout constraints. I tried the +2 +3 approach. the font still remains small  with hardly any noticeable difference on bigger devices

Reply:
I see. Will test your PR shortly and see how it works.